exp/controllers/awsmanagedcontrolplane_controller.go

```diff
@@ -0,0 +1,292 @@
+	if err := r.Client.Get(ctx, req.NamespacedName, awsControlPlane); err != nil {
+		if apierrors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	// Get the cluster
+	cluster, err := util.GetOwnerCluster(ctx, r.Client, awsControlPlane.ObjectMeta)
+	if err != nil {
+		logger.Error(err, "Failed to retrieve owner Cluster from the API Server")
+		return ctrl.Result{}, err
+	}
+	if cluster == nil {
+		logger.Info("Cluster Controller has not yet set OwnerRef")
+		return ctrl.Result{}, nil
+	}
+
+	if util.IsPaused(cluster, awsControlPlane) {
+		logger.Info("Reconciliation is paused for this object")
+		return ctrl.Result{}, nil
+	}
+
+	logger = logger.WithValues("cluster", cluster.Name)
+
+	managedScope, err := scope.NewManagedControlPlaneScope(scope.ManagedControlPlaneScopeParams{
+		Client:         r.Client,
+		Logger:         logger,
+		Cluster:        cluster,
+		ControlPlane:   awsControlPlane,
+		ControllerName: "awsmanagedcontrolplane",
+	})
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to create scope: %w", err)
+	}
+
+	// Always close the scope
+	defer func() {
+		applicableConditions := []clusterv1.ConditionType{
+			infrav1exp.EKSControlPlaneReadyCondition,
+			infrav1exp.IAMControlPlaneRolesReadyCondition,
+			infrav1.VpcReadyCondition,
+			infrav1.SubnetsReadyCondition,
+			infrav1.ClusterSecurityGroupsReadyCondition,
+		}
+
+		if managedScope.VPC().IsManaged(managedScope.Name()) {
+			applicableConditions = append(applicableConditions,
+				infrav1.InternetGatewayReadyCondition,
+				infrav1.NatGatewaysReadyCondition,
+				infrav1.RouteTablesReadyCondition,
+			)
+			if managedScope.Bastion().Enabled {
+				applicableConditions = append(applicableConditions, infrav1.BastionHostReadyCondition)
+			}
+		}
+
+		conditions.SetSummary(managedScope.ControlPlane, conditions.WithConditions(applicableConditions...), conditions.WithStepCounter())
+
+		if err := managedScope.Close(); err != nil && reterr == nil {
+			reterr = err
+		}
+	}()
+
+	if !awsControlPlane.ObjectMeta.DeletionTimestamp.IsZero() {
+		// Handle deletion reconciliation loop.
+		return r.reconcileDelete(ctx, managedScope)
+	}
+
+	// Handle normal reconciliation loop.
+	return r.reconcileNormal(ctx, managedScope)
+}
+
+func (r *AWSManagedControlPlaneReconciler) reconcileNormal(ctx context.Context, managedScope *scope.ManagedControlPlaneScope) (res ctrl.Result, reterr error) {
+	managedScope.Info("Reconciling AWSManagedControlPlane")
+
+	awsManagedControlPlane := managedScope.ControlPlane
+
+	controllerutil.AddFinalizer(managedScope.ControlPlane, infrav1exp.ManagedControlPlaneFinalizer)
+	if err := managedScope.PatchObject(); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	ec2Service := ec2.NewService(managedScope)
+	networkSvc := network.NewService(managedScope)
+	ekssvc := eks.NewService(managedScope)
+
+	if err := networkSvc.ReconcileNetwork(); err != nil {
```

Comment:
I wonder if ReconcileSecurityGroups needs to be pulled out of this method - each EKS cluster comes with a security group that gets applied to the underlying controlplane infrastructure, and that SG needs to be applied to all nodes. Because of that I'd expect ReconcileNetwork to assign the "controlplane" security group to THAT security group, as the one that ReconcileNetwork currently creates doesn't add any value as there aren't any control plane "instances" in the EKS case.

With a hacky workaround for #1852 and the EKS-created SG added to my AWSMachine's `additionalSecurityGroups`, I was actually able to use the EKS bootstrap provider and join a node. So we're really getting there!

Reply:
Thats a good point @rudoi . I will pull the `ReconcileSecurityGroups` out.

I will also add the id for the control plane security group to the status of the `AWSManagedControlPlane`. I can look this up by looking at the value of the `aws:eks:cluster-name` tag on the security group.

Reply:
I've extracted out the securitygroup reconciliation from the network service as part of #1857. Additional the latest commit will lookup the automatically created security group and adds the details to the status section